designs/lily/src/front.mjs

```diff
@@ -0,0 +1,610 @@
+
+  /*
+   * With the cross seams matched back and front,
+   * all that's left is to match the inseam and outseam
+   */
+
+  // When both are too short/long, adapt the leg length
+  if ((inseamDelta() < 0 && outseamDelta() < 0) || (inseamDelta() > 0 && outseamDelta() > 0))
+    adaptInseamAndOutseam()
+
+  // Now one is ok, the other will be adapted
+  adaptOutseam()
+  adaptInseam()
+
+  // Changing one will ever so slightly impact the other, so let's run both again to be sure
+  adaptOutseam()
+  adaptInseam()
+
+  // Only now style the waist lower if requested
+  // Note: redo this for lily even though it was already done for titan;
+  //  calculation for titan happened using its own seam lengths
+  if (options.waistHeight < 1 || absoluteOptions.waistbandWidth > 0) {
+    points.styleWaistOutLily = drawOutseam().shiftAlong(
+      measurements.waistToHips * (1 - options.waistHeight) + absoluteOptions.waistbandWidth
+    )
+    points.styleWaistInLily = utils.beamsIntersect(
+      points.styleWaistOutLily,
+      points.styleWaistOutLily.shift(points.waistOut.angle(points.waistIn), 10),
+      points.waistIn,
+      points.crotchSeamCurveStart
+    )
+  } else {
+    points.styleWaistInLily = points.waistIn.clone()
+    points.styleWaistOutLily = points.waistOut.clone()
+  }
+  let test = points.styleWaistInLily.dist(points.styleWaistOutLily)
+  console.log('front waist length', test)
+  store.set('frontWaist', points.styleWaistInLily.dist(points.styleWaistOutLily))
+
+  // Seamline
+  paths.seam = drawPath().attr('class', 'fabric')
+
+  // adjust the length (at the bottom)
+  let extendBeyondKnee = 1
+  if (options.lengthReduction > 0) {
+    let requestedLength = store.get('requestedLength')
+    // leggings must reach to fork at least, so define a minimum
+    let waistToFork = points.waistX.dy(points.fork)
+    let waistToKnee = points.waistX.dy(points.knee) // adapting the seams may have shifted the knee up or down
+    if (waistToFork > requestedLength) {
+      //log.warning('length reduction capped; cutting off at fork') // log only for back part
+      // add one percent to waistToFork to ensure that path length is nonzero
+      requestedLength = waistToFork * 1.01
+    }
+
+    // work-around to avoid splitting exactly at the knee
+    //   (due to a bug, splitting a path at a node is not possible)
+    if (0.999 < requestedLength / waistToKnee && requestedLength / waistToKnee < 1.001) {
+      requestedLength = 1.001 * waistToKnee
+    }
+    points.bottom = points.waistX.shift(270, requestedLength)
+    let upperPoint, upperCp
+    if (requestedLength < waistToKnee) {
+      extendBeyondKnee = 0
+
+      // 'cut' between fork and knee
+      if (points.waistOut.x < points.seatOut.x) {
+        upperPoint = points.styleWaistOutLily
+        upperCp = points.seatOut
+      } else {
+        upperPoint = points.seatOut
+        upperCp = points.seatOutCp2
+      }
+      points.bottomOut = utils.lineIntersectsCurve(
```

Comment:
Don't use `999` to extend a line, use a beam instead (`beamInstersecsCurve`)

Reply:
I've updated it to use beamIntersectsCurve, but since it's a single point that I'm using to define a horizontal line/beam, the 999 is still there.